src/main/java/picard/fingerprint/FingerprintUtils.java

```diff
@@ -137,15 +152,38 @@ private static VariantContext getVariantContext(final ReferenceSequenceFile refe
                 snp.getPos(),
                 snp.getPos()).getBases()[0]);
 
-        final Allele allele1 = Allele.create(snp.getAllele1(), snp.getAllele1() == refAllele);
-        final Allele allele2 = Allele.create(snp.getAllele2(), snp.getAllele2() == refAllele);
-        final List<Allele> alleles = Arrays.asList(allele1, allele2);
+        if (snp.getAllele1() != refAllele && snp.getAllele2() != refAllele){
+            throw new PicardException("Don't know how to deal with missing reference allele in fingerprinting map");
+        }
+
+        final Allele alleleRef;
+        final Allele alleleAlt;
+        final int obsRef, obsAlt;
+        final boolean swap12 = snp.getAllele2() == refAllele;
+
+        if (swap12) {
+            alleleRef = Allele.create(snp.getAllele2(), true);
+            alleleAlt = Allele.create(snp.getAllele1(), false);
+            obsRef = haplotypeProbabilities.getObsAllele2();
+            obsAlt = haplotypeProbabilities.getObsAllele1();
+        } else {
+            alleleRef = Allele.create(snp.getAllele1(), true);
+            alleleAlt = Allele.create(snp.getAllele2(), false);
+            obsRef = haplotypeProbabilities.getObsAllele1();
+            obsAlt = haplotypeProbabilities.getObsAllele2();
+        }
+
+        final double[] PLs =  Arrays.copyOf(haplotypeProbabilities.getLogLikelihoods(),HaplotypeProbabilities.NUM_GENOTYPES);
```

Comment:
```suggestion
        final double[] PLs =  Arrays.copyOf(haplotypeProbabilities.getLogLikelihoods(), HaplotypeProbabilities.NUM_GENOTYPES);
```

Reply:
Is there a reason to use `NUM_GENOTYPES` here instead of `haplotypeProbabilities.getLogLikelihoods().length`? Using `length` would make it clearer what the code is doing; it's copying the entire array.

Reply:
shorter? but OK.